ignite/contrib/metrics/fid.py

```diff
@@ -0,0 +1,89 @@
+import os
+from typing import Callable, Sequence, Union
+
+import torch
+import torch.nn as nn
+from torchvision import transforms
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+try:
+    from scipy import linalg
+except ImportError:
+    raise RuntimeError("This contrib module requires scipy to be installed.")
+
+
+__all__ = ["FID"]
+
+
+class FID(Metric):
+    """Calculates FID metric
+
+    """
+
+    def __init__(
+        self, output_transform: Callable = lambda x: x, fid_model: nn.Module = None,
+    ):
+        if fid_model is None:
+            try:
+                from torchvision import models
+
+                fid_model = models.inception_v3(pretrained=True, transform_input=True)
+            except ImportError:
+                raise ValueError("Argument fid_model should be set")
+
+        self._device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self._fid_model = fid_model.eval().to(self._device)
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._num_of_data = 0
+
+        self._features_sum_real = torch.tensor(0, device=self._device, dtype=torch.float32)
+        self._features_sum_fake = torch.tensor(0, device=self._device, dtype=torch.float32)
+
+        self._cov_real = torch.tensor(0, device=self._device, dtype=torch.float32)
+        self._cov_fake = torch.tensor(0, device=self._device, dtype=torch.float32)
+
+    @reinit__is_reduced
+    def update(self, output) -> None:
+        y_pred, y = output[0].detach(), output[1].detach()
+
+        with torch.no_grad():
+            batch_features_real = self._fid_model(y).sum(axis=0)
+            batch_features_fake = self._fid_model(y_pred).sum(axis=0)
+
+        self._num_of_data += 1
+
+        self._features_sum_real = self._features_sum_real + batch_features_real
+        self._features_sum_fake = self._features_sum_fake + batch_features_fake
+
+        if self._num_of_data > 1:
```

Comment:
Let's replace it like
```python
if self._cov_real is None:
    self._cov_real = torch.eye(len(y))
else:
     self._cov_real = self._compute_cov(batch_features_real, self._features_sum_real)
```
where we can refactor cov computation into `self._compute_cov` static function. Same for fake.